# Lesson 12 Exercise

In [3]:
library(ggplot2)
library(car)

We're going to work with some data included in R on diamonds.  Let's explore it a bit:

In [5]:
data(diamonds)

df <- diamonds

df[1:10, ]

carat cut       color clarity depth table price x    y    z   
1  0.23  Ideal     E     SI2     61.5  55    326   3.95 3.98 2.43
2  0.21  Premium   E     SI1     59.8  61    326   3.89 3.84 2.31
3  0.23  Good      E     VS1     56.9  65    327   4.05 4.07 2.31
4  0.29  Premium   I     VS2     62.4  58    334   4.20 4.23 2.63
5  0.31  Good      J     SI2     63.3  58    335   4.34 4.35 2.75
6  0.24  Very Good J     VVS2    62.8  57    336   3.94 3.96 2.48
7  0.24  Very Good I     VVS1    62.3  57    336   3.95 3.98 2.47
8  0.26  Very Good H     SI1     61.9  55    337   4.07 4.11 2.53
9  0.22  Fair      E     VS2     65.1  61    337   3.87 3.78 2.49
10 0.23  Very Good H     VS1     59.4  61    338   4.00 4.05 2.39

We've all probably heard about "carats" so let's see if it is associated with cost:

In [7]:
mod <- lm(price ~ carat, df)
summary(mod)


Call:
lm(formula = price ~ carat, data = df)

Residuals:
     Min       1Q   Median       3Q      Max 
-18585.3   -804.8    -18.9    537.4  12731.7 

Coefficients:
            Estimate Std. Error t value Pr(>|t|)    
(Intercept) -2256.36      13.06  -172.8   <2e-16 ***
carat        7756.43      14.07   551.4   <2e-16 ***
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Residual standard error: 1549 on 53938 degrees of freedom
Multiple R-squared:  0.8493,	Adjusted R-squared:  0.8493 
F-statistic: 3.041e+05 on 1 and 53938 DF,  p-value: < 2.2e-16


Seems like carats have a big effect, but we should check the units:

In [8]:
summary(df$carat)

   Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
 0.2000  0.4000  0.7000  0.7979  1.0400  5.0100 

It looks like carats range from .2 to 5.01.  We should probably talk about .1 unit changes instead of 1 unit changes since a 1 carat jump seems like a lot:

For a .1 carat increase, we expect the price to increase by $775.64

Let's add another term to the model - how about clarity:

In [12]:
df$clarity <- factor(df$clarity, ordered = FALSE)

mod <- lm(price ~ carat + clarity, df)
summary(mod)


Call:
lm(formula = price ~ carat + clarity, data = df)

Residuals:
     Min       1Q   Median       3Q      Max 
-17355.1   -639.9   -110.6    480.4  11162.8 

Coefficients:
            Estimate Std. Error t value Pr(>|t|)    
(Intercept) -6911.57      50.22 -137.61   <2e-16 ***
carat        8440.06      12.65  667.13   <2e-16 ***
claritySI2   2879.18      49.47   58.20   <2e-16 ***
claritySI1   3729.45      49.16   75.86   <2e-16 ***
clarityVS2   4388.90      49.38   88.88   <2e-16 ***
clarityVS1   4613.76      50.13   92.03   <2e-16 ***
clarityVVS2  5163.32      51.62  100.02   <2e-16 ***
clarityVVS1  5186.62      53.05   97.77   <2e-16 ***
clarityIF    5513.14      57.37   96.11   <2e-16 ***
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Residual standard error: 1294 on 53931 degrees of freedom
Multiple R-squared:  0.8949,	Adjusted R-squared:  0.8948 
F-statistic: 5.737e+04 on 8 and 53931 DF,  p-value: < 2.2e-16


In [11]:
summary(df$clarity)

I1   SI2   SI1   VS2   VS1  VVS2  VVS1    IF 
  741  9194 13065 12258  8171  5066  3655  1790

There appear to be 8 levels of clarity - this is a categorical variable.

Interpret the value of clarityVS1:

Being in clarity category VS1 is associated with a $4,613.76 increase in diamond price when compared to category I1. 

Now let's consider what to do when all categories are not "significant":

In [13]:
df$color <- factor(df$color, ordered = FALSE)

mod <- lm(price ~ clarity + color, df)
summary(mod)


Call:
lm(formula = price ~ clarity + color, data = df)

Residuals:
   Min     1Q Median     3Q    Max 
 -6046  -2469  -1308   1164  16858 

Coefficients:
             Estimate Std. Error t value Pr(>|t|)    
(Intercept)  2853.080    149.980  19.023  < 2e-16 ***
claritySI2   1341.922    147.882   9.074  < 2e-16 ***
claritySI1    264.764    146.258   1.810  0.07026 .  
clarityVS2    217.599    146.516   1.485  0.13751    
clarityVS1     -9.579    148.525  -0.064  0.94858    
clarityVVS2  -393.844    152.377  -2.585  0.00975 ** 
clarityVVS1 -1243.262    155.998  -7.970 1.62e-15 ***
clarityIF   -1020.983    169.235  -6.033 1.62e-09 ***
colorE          9.419     61.245   0.154  0.87777    
colorF        714.648     61.726  11.578  < 2e-16 ***
colorG       1094.251     60.167  18.187  < 2e-16 ***
colorH       1419.952     63.521  22.354  < 2e-16 ***
colorI       2033.169     70.684  28.764  < 2e-16 ***
colorJ       2185.993     86.997  25.127  < 2e-16 ***
---
Signif. codes:  0 ‘***’ 0.001 ‘

In [14]:
summary(df$color)

D     E     F     G     H     I     J 
 6775  9797  9542 11292  8304  5422  2808

Now we can see that some colors are significant, but E is not.  That means that color E is not sigificantly different than color D (the reference class).  But, we can see that the other color levels are significantly different.  

In practice, we have justification to leave this variable in the model.